baseapp/baseapp_test.go

```diff
@@ -2153,3 +2201,140 @@ func TestBaseApp_Init(t *testing.T) {
 		require.Equal(t, tc.expectedSnapshot.KeepRecent, tc.bapp.snapshotManager.GetKeepRecent())
 	}
 }
+
+func executeBlockWithArbitraryTxs(t *testing.T, app *BaseApp, numTransactions int, blockHeight int64) []txTest {
+	codec := codec.NewLegacyAmino()
+	registerTestCodec(codec)
+	r := rand.New(rand.NewSource(randSource))
+	randSource *= 2
+	keyCounter := r.Intn(100)
+	txs := make([]txTest, 0)
+
+	app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: blockHeight}})
+	for txNum := 0; txNum < numTransactions; txNum++ {
+		tx := txTest{Msgs: []sdk.Msg{}}
+		for msgNum := 0; msgNum < 2; msgNum++ {
+			key := []byte(fmt.Sprintf("%v", keyCounter))
+			value := make([]byte, 10000)
+			_, err := r.Read(value)
+			require.NoError(t, err)
+			tx.Msgs = append(tx.Msgs, msgKeyValue{Key: key, Value: value})
+			keyCounter++
+		}
+		txBytes, err := codec.Marshal(tx)
+		require.NoError(t, err)
+		resp := app.DeliverTx(abci.RequestDeliverTx{Tx: txBytes})
+		require.True(t, resp.IsOK(), "%v", resp.String())
+		txs = append(txs, tx)
+	}
+	app.EndBlock(abci.RequestEndBlock{Height: blockHeight})
+	return txs
+}
+
+func executeBlock(t *testing.T, app *BaseApp, txs []txTest, blockHeight int64) {
+	codec := codec.NewLegacyAmino()
+	registerTestCodec(codec)
+	numTransactions := len(txs)
+
+	app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: blockHeight}})
+	for txNum := 0; txNum < numTransactions; txNum++ {
```

Comment:
Can you do `for _, tx := range txs` instead?

Reply:
Updated